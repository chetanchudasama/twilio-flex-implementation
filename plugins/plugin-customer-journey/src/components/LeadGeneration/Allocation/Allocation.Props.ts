export interface OwnProps {
  getRankedAgents: () => void;
}

export type AllocationProps = OwnProps;
