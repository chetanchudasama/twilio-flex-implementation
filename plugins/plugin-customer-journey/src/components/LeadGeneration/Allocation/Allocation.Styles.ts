import { black } from "@common/components";
import styled from "react-emotion";

export const AllocationStyles = styled("div")({
  padding: 24,
  width: "100%",
  ".heading": {
    fontSize: 20,
    lineHeight: "24px",
    fontWeight: 600,
    color: black,
  },
  ".allocation": {
    display: "flex",
    alignItems: "center",
    justifyContent: "space-between",
    padding: "2px 2px 2px 8px !important",
    marginLeft: -8,
  },
  ".find-agent-btn": {
    borderRadius: 4,
    fontWeight: 500,
    fontSize: 14,
    height: 40,
    boxShadow: "0px 1px 2px 0px #2C2B350A",
  },
});
