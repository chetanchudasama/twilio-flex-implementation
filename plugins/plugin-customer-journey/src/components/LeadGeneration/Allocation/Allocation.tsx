import React from "react";
import { Button, Grid } from "@material-ui/core";
import { AllocationStyles } from "./Allocation.Styles";
import { AllocationProps } from "./Allocation.Props";

export const Allocation: React.FC<AllocationProps> = ({ getRankedAgents }) => {
  return (
    <AllocationStyles>
      <Grid container spacing={16}>
        <Grid item xs={12} className="allocation">
          <div className="heading">Allocation</div>
          <div>
            <Button
              variant="contained"
              className="find-agent-btn"
              color="secondary"
              onClick={getRankedAgents}
            >
              Find Agent
            </Button>
          </div>
        </Grid>
      </Grid>
    </AllocationStyles>
  );
};
