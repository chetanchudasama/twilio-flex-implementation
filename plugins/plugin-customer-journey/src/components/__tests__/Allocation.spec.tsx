import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";

import { Allocation } from "../LeadGeneration/Allocation/Allocation";
import { AllocationProps } from "../LeadGeneration/Allocation/Allocation.Props";

const props: AllocationProps = {
  getRankedAgents: jest.fn(),
};

describe("Allocation", () => {
  it("renders correctly on load", () => {
    const wrapper = create(<Allocation {...props} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it("should render Find Agent button correctly", () => {
    const wrapper = mount(<Allocation {...props} />);

    const button = wrapper.find(".find-agent-btn").last();
    expect(button.exists()).toBeTruthy();
  });

  it("should emit getRankedAgents method, when find agent button is clicked", () => {
    const wrapper = mount(<Allocation {...props} />);

    const button = wrapper.find(".find-agent-btn").last();
    button.simulate("click");

    expect(props.getRankedAgents).toHaveBeenCalled();
  });
});
