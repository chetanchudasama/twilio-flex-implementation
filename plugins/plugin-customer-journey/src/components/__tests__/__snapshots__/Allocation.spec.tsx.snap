// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Allocation renders correctly on load 1`] = `
<div
  className="css-18isgbz"
>
  <div
    className="MuiGrid-container-1 MuiGrid-spacing-xs-16-24"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 allocation"
    >
      <div
        className="heading"
      >
        Allocation
      </div>
      <div>
        <button
          className="MuiButtonBase-root-124 MuiButton-root-98 MuiButton-contained-109 MuiButton-containedSecondary-111 MuiButton-raised-112 MuiButton-raisedSecondary-114 find-agent-btn"
          disabled={false}
          onBlur={[Function]}
          onClick={[MockFunction]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-99"
          >
            Find Agent
          </span>
          <span
            className="MuiTouchRipple-root-127"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
